src/test/java/com/google/maps/PlacesApiTest.java

```diff
@@ -412,7 +419,29 @@ public void testTextSearchRequest() throws Exception {
     assertParamValue(String.valueOf(3000), "radius", actualParams);
     assertParamValue(String.valueOf(1), "minprice", actualParams);
     assertParamValue(String.valueOf(4), "maxprice", actualParams);
+    assertParamValue("name", "name", actualParams);
     assertParamValue("true", "opennow", actualParams);
+    assertParamValue(Rankby.DISTANCE.toString(), "rankby", actualParams);
+    assertParamValue(PlaceType.AIRPORT.toString(), "type", actualParams);
+  }
+
+  @Test
+  public void testTextSearchLocationWithoutRadius() throws Exception {
+    MockResponse response = new MockResponse();
+    response.setBody("");
+    server.enqueue(response);
+    server.play();
+    context.setBaseUrlForTesting("http://127.0.0.1:" + server.getPort());
+
+    LatLng location = new LatLng(10, 20);
+    try {
+      PlacesApi.textSearchQuery(context, "query")
```

Comment:
You can drop the try/catch block here (and elsewhere) and use the annotation @Test(expected=IllegalArgumentException.class) instead.